buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -664,39 +671,45 @@ def set_test_targets() {
 * Set EXTRA_TEST_LABELS map if defined in variable file.
 */
 def set_test_misc() {
-    EXTRA_TEST_LABELS = [:]
     if (!params.TEST_NODE) {
         // Only add extra test labels if the user has not specified a specific TEST_NODE
-        TARGET_NAMES.each { target ->
-            EXTRA_TEST_LABELS[target] = buildspec.getVectorField("extra_test_labels", target).join('&&') ?: ''
+        TESTS.each { id, target ->
+            target['extraTestLabels'] = buildspec.getVectorField("extra_test_labels", target).join('&&') ?: ''
         }
     }
 
-    EXCLUDED_TESTS = []
     def buildspec = buildspec_manager.getSpec(SPEC)
     def excludedTests = buildspec.getVectorField("excluded_tests", SDK_VERSION)
     if (excludedTests) {
-        EXCLUDED_TESTS.addAll(excludedTests)
+        excludedTestsArray = []
+        excludedTestsArray.addAll(excludedTests)
+        TESTS.each { id, target ->
+            if (excludedTestsArray.contains(id)) {
+                echo "Excluding test target:'${id}'"
+                TESTS.remove(id)
+            }
+        }
     }
 
-    TEST_FLAG = buildspec.getScalarField("test_flags", SDK_VERSION) ?: ''
+    TESTS.each { id, target ->
+        flag = buildspec.getScalarField("test_flags", id) ?: ''
+        if (flag) {
+            target['testFlag'] = (target['testFlag']) ? error("Cannot set more than 1 TEST_FLAG") : flag
+        } else {
+            target['testFlag'] = (target['testFlag']) ? target['testFlag'] : ''
+        }
+    }
 
     // Set test param KEEP_REPORTDIR to false unless set true in variable file.
-    TEST_KEEP_REPORTDIR = [:]
-    TARGET_NAMES.each { target ->
-        TEST_KEEP_REPORTDIR[target] = buildspec_manager.getSpec('misc').getScalarField("test_keep_reportdir", target) ?: 'false'
+    TESTS.each { id, target ->
```

Comment:
Why another loop here for `keepReportDir`? Set `target['keepReportDir'] `in the above loop.